qiskit/qpy/binary_io/circuits.py

```diff
@@ -272,6 +316,22 @@ def _parse_custom_instruction(custom_instructions, gate_name, params):
         inst_obj.definition = definition
         return inst_obj
 
+    if type_key == common.CircuitInstructionTypeKey.CONTROLLED_GATE:
```

Comment:
Just to confirm, it's never reached with version < 4 since controlled gate doesn't appear in QPY data?

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/8055/commits/dd25b3853fbdb6864bd86c6d1f71e38e7e451e12